harness/determined/layers/_workload_manager.py

```diff
@@ -141,6 +140,12 @@ def _respond(in_response: workload.Response) -> None:
 
             in_response = cast(workload.Metrics, in_response)
             metrics = in_response["metrics"]
+
+            metrics_size = det.util.sizeof_dict(metrics)
```

Comment:
blocking: I tried to time json-encoding 100MB object, which takes 27ms. It doesn't hurt much but I'd check the size in `_socket_manager.py`, where calls `json_encode`. Then you don't need to encode the metrics twice unnecessarily.

Reply:
the response in _socket_manager.py includes other data that isn't directly what's recorded on a column-basis to the database. it includes some overhead for other columns: 
`{'type': 'WORKLOAD_COMPLETED', 'workload': <RUN_STEP (100 Batches): (1,1,10)>, 'start_time': datetime.datetime(2021, 5, 17, 19, 35, 34, 97884, tzinfo=datetime.timezone.utc), 'end_time': datetime.datetime(2021, 5, 17, 19, 35, 37, 558983, tzinfo=datetime.timezone.utc), 'metrics': {...metrics }`
not a huge deal, but not 100% accurate in terms of the size of the object being persisted to the column. i guess i could encode the actual metrics piece of that json separately, then manually stitch it together to send in _socket_manager respond, but that seems confusing from a code perspective. what do you think?